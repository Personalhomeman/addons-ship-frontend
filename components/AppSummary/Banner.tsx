export type Props = {
  startColor: string;
  endColor: string;
};

export default ({ startColor, endColor }: Props) => (
  <svg
    viewBox="0 0 862 80"
    style={{
      background: `linear-gradient(90deg, ${startColor}, ${endColor})`
    }}
  >
    <defs>
      <linearGradient x1="96.609239%" y1="49.6431115%" x2="14.0733222%" y2="50.6271256%" id="linearGradient-1">
        <stop stopColor="#FFFFFF" stopOpacity="0" offset="0%"></stop>
        <stop stopColor="#FFFFFF" stopOpacity=".5" offset="64.6547264%"></stop>
        <stop stopColor="#D0FEFF" stopOpacity="0" offset="100%"></stop>
      </linearGradient>
      <linearGradient x1="96.609239%" y1="49.381157%" x2="-22.6276744%" y2="52.5123108%" id="linearGradient-2">
        <stop stopColor="#FFFFFF" stopOpacity="0" offset="0%"></stop>
        <stop stopColor="#FFFFFF" stopOpacity=".8" offset="47.0001462%"></stop>
        <stop stopColor="#D0FEFF" stopOpacity="0" offset="100%"></stop>
      </linearGradient>
      <linearGradient x1="88.6366729%" y1="49.8847704%" x2="14.0733222%" y2="50.3332489%" id="linearGradient-3">
        <stop stopColor="#FFFFFF" stopOpacity="0" offset="0%"></stop>
        <stop stopColor="#FFFFFF" stopOpacity=".2" offset="52.3083683%"></stop>
        <stop stopColor="#FFFFFF" stopOpacity="0" offset="100%"></stop>
      </linearGradient>
    </defs>
    <g id="1st-Iteration---Without-Testing,-QA,-Dashboard" stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
      <g id="Latest-App-Copy-2" transform="translate(-91.000000, 0.000000)">
        <g id="Group-3">
          <g id="Group-8">
            <g id="Group-6" transform="translate(-15.999624, -6.999236)">
              <path
                d="M881.735481,86.9990641 L210.156384,86.9978664 C218.71153,52.9875994 258.306013,22.8146204 333.594585,9.30207949 C448.470008,-11.3153767 695.406951,42.43158 881.735481,86.9990641 Z"
                id="Combined-Shape"
                fill-opacity="0.5"
                fill="url(#linearGradient-1)"
                opacity="0.430636"
                transform="translate(545.945975, 45.880259) scale(-1, 1) translate(-545.945975, -45.880259) "
              ></path>
              <path
                d="M975.999624,7.00298921 L975.999624,86.9972046 L479.881688,87.0001375 C522.81394,59.2648802 567.029696,31.8415792 610.365803,7.00153703 L975.999624,7.00298921 Z"
                id="Combined-Shape"
                fill-opacity="0.5"
                fill="url(#linearGradient-2)"
                opacity="0.430636"
              ></path>
              <path
                d="M871.7687,87.0029988 L0.532462766,86.9976624 C26.9483975,25.2137692 111.220589,-23.3393292 246.79476,14.1075744 C423.760964,62.9873598 649.806848,31.4599363 798.08942,7.20277579 L963.958206,7.20043089 C965.226263,31.2899681 937.377576,70.049062 871.7687,87.0029988 Z"
                id="Combined-Shape"
                fill="url(#linearGradient-3)"
                transform="translate(482.265999, 44.001117) scale(-1, 1) translate(-482.265999, -44.001117) "
              ></path>
            </g>
          </g>
        </g>
      </g>
    </g>
  </svg>
);
