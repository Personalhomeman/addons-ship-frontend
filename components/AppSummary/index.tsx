import css from './style.scss';
import SVG from 'react-svg';
import Link from 'next/link';

interface AppSummaryProps {
  detailsPagePath: string;
  title: string;
  description: string;
  note: string;
  iconUrl: string;
  platformIconUrl: string;
}

export default ({ detailsPagePath, title, description, note, iconUrl, platformIconUrl }: AppSummaryProps) => (
  <Link href={detailsPagePath}>
    <a className={css.appSummary}>
      <div className={css.colorBar} />
      <div className={css.body}>
        <div className={css.iconWrapper}>
          <div className={css.icon}>
            <SVG src={iconUrl} />
          </div>
          <div className={css.note}>{note}</div>
        </div>
        <div className={css.titleWrapper}>
          <SVG src={platformIconUrl} />
          <div className={css.title}>{title}</div>
        </div>
        <div className={css.description}>{description}</div>
      </div>
    </a>
  </Link>
);
