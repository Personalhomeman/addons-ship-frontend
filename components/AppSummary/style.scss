@import '~@/assets/style/mixins';
@import '~@/assets/style/dimensions';
@import '~@/assets/style/colors';

.appSummary {
  $hover-border-width: 1px;

  display: block;
  border: $hover-border-width solid $gray-2;
  border-radius: 4 * $grid;
  overflow: hidden;
  background-color: $white;
  text-decoration: none;
  // TODO: replace with element from UI library
  box-shadow: 0 2px 12px 0 rgba(0, 0, 0, 0.06);

  &:hover,
  &:focus {
    border-color: $grape-3;
  }

  .colorBar {
    height: 20 * $grid - $hover-border-width;
    background: conic-gradient(from 266deg at 100% 50%, rgba(0, 0, 0, 0), rgba(255, 255, 255, 0.2)),
      linear-gradient(to right, rgba(13, 131, 205, 1), rgba(29, 227, 219, 1));
  }

  .body {
    padding: 8 * $grid 8 * $grid - $hover-border-width;

    .iconWrapper {
      display: flex;
      align-items: flex-end;
      margin-top: -20 * $grid;

      .icon {
        @include size(24 * $grid);
        border-radius: 4 * $grid;
      }

      .note {
        margin-left: 4 * $grid;
        margin-bottom: 2 * $grid;
        color: $gray-6;
        // TODO: replace with element from UI library
        line-height: 1.5;
        letter-spacing: normal;
        font-weight: 500;
        font-style: normal;
        font-stretch: normal;
        font-size: 16px;
      }
    }

    .titleWrapper {
      display: flex;
      align-items: center;
      margin-top: 4 * $grid;

      svg {
        display: block;
        fill: $grape-5;
      }

      .title {
        margin-left: 2 * $grid;
        color: $grape-5;
        // TODO: replace with element from UI library
        line-height: 1.5;
        letter-spacing: normal;
        font-style: normal;
        font-stretch: normal;
        font-size: 25px;
        font-weight: 700;
      }
    }

    .description {
      margin-top: 2 * $grid;
      color: $grape-5;
      // TODO: replace with element from UI library
      line-height: 1.5;
      letter-spacing: normal;
      font-weight: 500;
      font-style: normal;
      font-stretch: normal;
      font-size: 16px;
    }
  }
}
