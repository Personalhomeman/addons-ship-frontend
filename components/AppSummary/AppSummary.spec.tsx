import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import AppSummary from '.';

describe('AppSummary', () => {
  it('renders correctly', () => {
    const tree = toJSON(
      shallow(
        <AppSummary
          title="My app v1.1.0 (28)"
          description="This is a short description about my app. It is a good app. End of description."
          note="Updated on January 29, 2008"
          iconUrl="/static/icon-flutter.svg"
          platformIconUrl="/static/icon-apple.svg"
          detailsPagePath="/details"
        />
      )
    );
    expect(tree).toMatchSnapshot();
  });
});
