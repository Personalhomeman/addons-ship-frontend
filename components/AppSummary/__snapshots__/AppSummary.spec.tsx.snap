// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSummary renders correctly 1`] = `
<Link
  href="/details"
>
  <a
    className="appSummary"
  >
    <div
      className="colorBar"
    />
    <div
      className="body"
    >
      <div
        className="iconWrapper"
      >
        <div
          className="icon"
        >
          <ReactSVG
            afterInjection={[Function]}
            beforeInjection={[Function]}
            evalScripts="never"
            fallback={null}
            loading={null}
            renumerateIRIElements={true}
            src="/static/icon-flutter.svg"
            wrapper="div"
          />
        </div>
        <div
          className="note"
        >
          Updated on January 29, 2008
        </div>
      </div>
      <div
        className="titleWrapper"
      >
        <ReactSVG
          afterInjection={[Function]}
          beforeInjection={[Function]}
          evalScripts="never"
          fallback={null}
          loading={null}
          renumerateIRIElements={true}
          src="/static/icon-apple.svg"
          wrapper="div"
        />
        <div
          className="title"
        >
          My app v1.1.0 (28)
        </div>
      </div>
      <div
        className="description"
      >
        This is a short description about my app. It is a good app. End of description.
      </div>
    </div>
  </a>
</Link>
`;
